.github/workflows/build_library.yml

```diff
@@ -10,31 +10,54 @@ env:
 
 jobs:
   linux:
+    timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
         build-arch:
-          - manylinux2010_i686
-          - manylinux2010_x86_64
-          - manylinux2014_aarch64
+          - { image: 'manylinux2014_x86_64', target: 'i686' }
+          - { image: 'manylinux2014_x86_64', target: 'x86_64' }
+          - { image: 'manylinux2014_x86_64', target: 'aarch64' }
 
-    name: Python Linux ${{ matrix.build-arch }}
+    name: Python Linux ${{ matrix.build-arch.target }}
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
 
-      - if: matrix.build-arch == 'manylinux2014_aarch64'
+      - if: matrix.build-arch.target == 'aarch64'
         uses: docker/setup-qemu-action@v1
         with:
           platforms: arm64
 
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and export
+        uses: docker/build-push-action@v3
+        with:
+          context: py/
+          tags: relay-cabi-builder-${{matrix.build-arch.target}}:latest
+          build-args: |
+            IMAGE=quay.io/pypa/${{matrix.build-arch.image}}
+            TARGET=${{matrix.build-arch.target}}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          outputs: type=docker,dest=/tmp/relay-cabi-builder-${{matrix.build-arch.target}}.tar
```

Comment:
I was not able to find the `outputs` and following load mentioned in the cache examples from this action -- is this indeed required with the `gha` cache type?

Reply:
as discussed I've removed these steps and build the image in the script instead